sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -675,6 +676,20 @@ object PushPredicateThroughGenerate extends Rule[LogicalPlan] with PredicateHelp
 }
 
 /**
+ * Push [[Filter]] operators through [[Aggregate]] operators in case predicate's
+ * refrence attributes are subset of groupBy attribute set.
+ */
+object PushPredicateThroughAggregate extends Rule[LogicalPlan] with PredicateHelper {
+
+  def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+    case filter @ Filter(condition,
+    aggregate @ Aggregate(groupingExpressions, aggregateExpressions, child))
+        if (filter.condition.references -- AttributeSet(groupingExpressions)).isEmpty =>
```

Comment:
rather than checking for complete overlap, can we pull out the expressions for group by columns and push those down?